core/src/main/java/jenkins/tasks/SimpleBuildWrapper.java

```diff
@@ -64,18 +64,69 @@
 @SuppressWarnings("rawtypes") // inherited
 public abstract class SimpleBuildWrapper extends BuildWrapper {
 
+    /**
+     * Determines whether or not this wrapper requires a workspace context (working directory and launcher).
+     * <p>
+     * When such a context is required (the default), {@link #setUp(Context, Run, FilePath, Launcher, TaskListener, EnvVars)} applies.
+     * Otherwise, {@link #setUp(Context, Run, TaskListener, EnvVars)} applies.
+     *
+     * @return {@code true} if this wrapper requires a workspace context; {@code false} otherwise.
+     * @since TODO
+     */
+    boolean requiresWorkspace() {
```

Comment:
```suggestion
    protected boolean requiresWorkspace() {
```

as you will see when you try to write a test for this

Reply:
Right.

Reply:
Actually, no. This needs to be `public` - it's the method you override to show you don't need a workspace context, which is information the caller needs to call the right `setUp` overload.